src/Workspaces/Core/Portable/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -46,29 +46,22 @@ public static bool IsOpen(this Document document)
         /// </summary>
         public static async Task<SemanticModel> GetSemanticModelForSpanAsync(this Document document, TextSpan span, CancellationToken cancellationToken)
         {
-            try
+            var syntaxFactService = document.GetLanguageService<ISyntaxFactsService>();
+            var semanticModelService = document.Project.Solution.Workspace.Services.GetService<ISemanticModelService>();
+            if (semanticModelService == null || syntaxFactService == null)
             {
-                var syntaxFactService = document.GetLanguageService<ISyntaxFactsService>();
-                var semanticModelService = document.Project.Solution.Workspace.Services.GetService<ISemanticModelService>();
-                if (semanticModelService == null || syntaxFactService == null)
-                {
-                    return await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-                }
-
-                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-                var token = root.FindToken(span.Start);
-                if (token.Parent == null)
-                {
-                    return await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-                }
-
-                var node = token.Parent.AncestorsAndSelf().FirstOrDefault(a => a.FullSpan.Contains(span));
-                return await GetSemanticModelForNodeAsync(semanticModelService, syntaxFactService, document, node, span, cancellationToken).ConfigureAwait(false);
+                return await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             }
-            catch (Exception e) when (FatalError.ReportUnlessCanceled(e))
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(span.Start);
+            if (token.Parent == null)
             {
-                throw ExceptionUtilities.Unreachable;
+                return await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             }
+
+            var node = token.Parent.AncestorsAndSelf().FirstOrDefault(a => a.FullSpan.Contains(span));
+            return await GetSemanticModelForNodeAsync(semanticModelService, syntaxFactService, document, node, span, cancellationToken).ConfigureAwait(false);
```

Comment:
we are firing exception because node is null which can't happen unless span cover outside of root node. and that can only happen if text <-> tree round tripping is broken.